ginga/util/wcsmod.py

```diff
@@ -286,15 +304,14 @@ def realize_frame(self, data):
                              " from {}".format(self.coordframe.representation))
             rep = astropy.coordinates.UnitSphericalRepresentation(*data)
 
-        if hasattr(self.coordframe._set_data, '_set_data'):
+        if hasattr(self.coordframe, '_set_data'):
             self.coordframe._set_data(rep)
         else:
             self.coordframe._data = rep
-            self.coordframe._rep_cache[self.coordframe._data.__class__.__name__,
```

Comment:
I think you need to clear the `_rep_cache` here to make sure that there is no left over data in it.